Data migration: add doc for migration with binlog event filter and pt-ghost (#7258)
pingcap/docs: data-migration/migrate-with-pt-ghost.md

---
title: Continuous Replication from Databases that Use gh-ost or pt-osc
summary: Learn how to use DM to replicate incremental data from databases that use online DDL tools gh-ost or pt-osc
---

# Continuous Replication from Databases that Use gh-ost or pt-osc

> In production scenarios, table locking during DDL execution can block the reads or writes to and from the database to a certain extent. Therefore, online DDL tools are often used to execute DDLs to minimize the impact on reads and writes. Common DDL tools are [gh-ost](https://github.com/github/gh-ost) and [pt-osc](https://www.percona.com/doc/percona-toolkit/3.0/pt-online-schema-change.html).

Author: ```suggestion
In production scenarios, table locking during DDL execution can block the reads from or writes to the database to a certain extent. Therefore, online DDL tools are often used to execute DDLs to minimize the impact on reads and writes. Common DDL tools are [gh-ost](https://github.com/github/gh-ost) and [pt-osc](https://www.percona.com/doc/percona-toolkit/3.0/pt-online-schema-change.html).
```

---

# Continuous Replication from Databases that Use gh-ost or pt-osc

In production scenarios, table locking during DDL execution can block the reads from or writes to the database to a certain extent. Therefore, online DDL tools are often used to execute DDLs to minimize the impact on reads and writes. Common DDL tools are [gh-ost](https://github.com/github/gh-ost) and [pt-osc](https://www.percona.com/doc/percona-toolkit/3.0/pt-online-schema-change.html).

After migrating data from MySQL to TiDB by using DM, you can enbale `online-ddl` to allow collaboration of DM and gh-ost or pt-osc.

For the detailed replication instruction, refer to the following documents by scenarios:

- [Migrate MySQL Data of Less Than 1 TiB to TiDB](/data-migration/migrate-mysql-tidb-less-tb.md)
- [Migrate MySQL Data of More Than 1 TiB to TiDB](/data-migration/migrate-mysql-tidb-above-tb.md)
- [Migrate and Merge MySQL Shards of Less Than 1 TiB to TiDB](/data-migration/migrate-shared-mysql-tidb-less-tb.md)
- [Migrate and Merge MySQL Shards of More Than 1 TiB to TiDB](/data-migration/migrate-shared-mysql-tidb-above-tb.md)

## Enable online-ddl on DM

> When configuring the task configuration file of DM, set the global parameter `online-ddl` to `true`, as shown below:

Author: ```suggestion
In the task configuration file of DM, set the global parameter `online-ddl` to `true`, as shown below:
```
